src/rest.ts

```diff
@@ -56,28 +57,29 @@ export function makeUrl(scheme: string, host: string, port: string) {
   return `${scheme}://${host}${port ? ':' + port : ''}`;
 }
 
-type InfraError = {
-  status: string;
-  message: string;
-  details?: string;
-};
-
-export class SdkError extends Error implements InfraError {
+export class SdkError extends Error {
   status: string;
   message: string;
   details?: string;
+  problems?: Problem[];
   response: Response;
 
   constructor(body: any, response: Response) {
```

Comment:
We do not seem to need `body` downstream so it might be cleaner to destructure it to `{ details, message, problems }`

Reply:
My bad, I missed this comment and merged. Maybe in a next PR.